### PR TITLE
[Spec] perf: Fuse topk=1 target verify finalization

### PR DESCRIPTION
**Pull request by mattteochen**
This PR depends on https://github.com/sgl-project/sglang/pull/31465.

## Motivation

For greedy EAGLE topk=1 decoding, target verification currently launches a full-vocabulary `argmax`, the generic `VerifyTreeGreedy` kernel, and several small ATen kernels to assemble acceptance and draft-extend metadata. These sequential launches sit between the target-verify and draft-extend CUDA graphs and add avoidable launch overhead.

## Modifications

- Reuse #30947's split topk=1 argmax reduction for target logits.
- Add a Triton finalizer that performs chain verification and directly materializes `predict`, raw accepted-draft counts, bonus-inclusive accept lengths, accept indices, bonus tokens, updated sequence lengths, selected-row indices, and int64 draft input IDs.
- Select the fused path only for CUDA `EAGLE_VERIFY`, topk=1, all-greedy batches with supported tensor metadata; keep the existing fallbacks for other modes and platforms.
- Pass the fused outputs through the worker into draft extend, removing redundant eager setup kernels.
- Add exact fused-versus-eager coverage for all outputs across FP16, BF16, and FP32, plus a CUDA-graph benchmark.

## Accuracy Tests
GLM5.2-FP8 B300, eval unchanged from main:
```bash
== gsm8k ==
1319 examples (single-shot)  |  536.9s  |  1970 tok/s  |  1.1M tokens

* score           =  94.69%
  stop_rate       =  96.74%
  truncated_rate  =  3.26%  [warn: hitting max_tokens]
  error_rate      =  0.00%
```

## Speed Tests and Profiling

CUDA graph replay benchmark with FP32 logits shaped `[batch_size * 6, 154880]`. Values are median latency in microseconds (`speculative_num_steps = 5`, `speculative_num_draft_tokens = 6`).

| Batch size | Acceptance | Eager (us) | Fused (us) | Speedup |
|---:|:---:|---:|---:|---:|
| 1 | None | 31.202 | 4.406 | 7.08x |
| 1 | All | 32.299 | 4.661 | 6.93x |
| 8 | None | 38.690 | 7.384 | 5.24x |
| 8 | All | 39.317 | 7.659 | 5.13x |
| 32 | None | 68.638 | 25.361 | 2.71x |
| 32 | All | 68.670 | 25.609 | 2.68x |

In the GLM-5.2 decode trace, device operations strictly between `FutureMap.publish` and `_fused_dsa_draft_extend_metadata_kernel` drop from 13 to 8. The five removed launches previously computed `accept_lens - 1`, `arange + accept_lens - 1`, and `predict.to(int64)`. The fused target finalizer now emits the equivalent `num_correct_drafts`, `select_index`, and int64 `draft_input_ids` directly. FutureMap and DSA ownership remain unchanged.

main:
image

PR:
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #32040320708](https://github.com/sgl-project/sglang/actions/runs/32040320708)
Latest PR Test (Extra): :x: [Run #32040320633](https://github.com/sgl-project/sglang/actions/runs/32040320633)

### CODE REVIEW

**Comment by mattteochen** [comment]
Last commit on this branch integrated the broadcast policy added by https://github.com/sgl-project/sglang/pull/34238
